Authentication/Authentication/Program.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Reflection.Metadata.Ecma335;
+
+namespace Authentication
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var path = @"./Data/Users.txt";
+            string choice = "0";
+
+            CheckDuplicateUsers(path);
+
+            while (!choice.Equals("Exit", StringComparison.OrdinalIgnoreCase))
+            {
+                choice = GetUserChoice();
+                if (choice.Equals("Login", StringComparison.OrdinalIgnoreCase))
+                {
+                    try
+                    {
+                        Login(path);
+                    }
+                    catch (FileNotFoundException exception)
+                    {
+                        throw new UsersNotFoundException(exception.FileName);
+                    }
+                    finally
+                    {
+                        Console.WriteLine();
+                    }
+                }
+                else if (choice.Equals("Register", StringComparison.OrdinalIgnoreCase))
+                {
+                    Register(path);
+                    Console.WriteLine();
+                }
+                else
+                {
+                    Console.WriteLine("Invalid choice: {0}", choice);
+                    Console.WriteLine();
+                }
```

Comment:
Note the difference between: ` CheckDuplicateUsers(path);` and the while loop. While loop requires you to go deep into details and read through the whole function just to figure out what the intention was. Instead, use a meaningful function name to clarify the intention. So split this function (while) into another.